modules/ROOT/pages/apid-behavioral-headers.adoc

```diff
@@ -0,0 +1,98 @@
++
+* If you use this header together with MS2-Status-Code and the mocking service (based on the probability) determines that an error code should be returned, the mocking service uses the error code that is specified by MS2-Status-Code. Therefore, for MS2-Status-Code you must specify both a non-error code and an error code, with the two codes separated by a comma.
+* If you use this header together with MS2-Error-Level and the mocking service (based on the probability) determines that an error code should be returned, the mocking service uses one of the error codes that is in the range specified by MS2-Error-Level. If the mocking service determines that a non-error code should be returned, it uses the first non-error code that is defined for the method.
+
+MS2-Randomize:: When this header is sent as true, the mocking service auto-generates a payload response if no example is defined for the response.
++
+For example, suppose that an API specification contained the definitions of this type:
++
+----
+types:
+  Person:
+    properties:
+      id: integer
+      name: string
+----
++
+If you did not provide an example, but you used the header MS2-Randomize, set to `true`, with the mocking service, the payload of the returned message would contain an example with random values that matched the data types, like this:
++
+----
+{
+  "id": 98358,
+  "name": "OVCKr"
+}
+----
++
+Moreover, the header respects more detailed definition of types. For example, you might change your type definition to look like this:
++
+----
+types:
+  Person:
+    properties:
+      id:
+        type: integer
+        minimum: 3
+        maximum: 30
+        multipleOf: 3
+      name: string
+----
++
+If MS2-Randomize were set to `true`, the payload in the response from the mocking service might look like this, with the value of `id` conforming to the constraints:
++
+----
+{
+  "id": 9,
+  "name": "rul68"
+}
+----
+
+MS2-Status-Code:: Specifies the non-error code and the error code that the mocking service can choose from for the return message. Separate the two status codes with a comma. For example, you might want the mocking service to return a 201 message, if there is no error, or a 401 status code, if there is an error; in this case, you would specify `201,401` for the value.
++
+If you use this header and you do not provide a non-error status code, the mocking service throws an error if the API does not return an error. Also, if you use this header and you do not provide an error status code, the mocking service throws an error if the API returns an error.
++
+If you do not use this header, the mocking service behaves as follows:
++
+* If the API returns a non-error status code, the mocking service returns the first non-error status code that is listed for the method.
+* If the API returns an error status code, the mocking service returns the first error status code that is listed for the method.
```

Comment:
This is actually a behavior related to MS2-Error-Rate, not to MS2-Status-Code

Reply:
Here I am saying what happens if the user does _not_ use MS2-Status-Code. I thought you said yesterday that the default behavior of the mocking service is to return the first defined non-error code (if there is no error) or the first defined error code (if there is an error). The point of MS2-Status-Code, I thought, was to override this default behavior.

Reply:
Ok, what I meant was that if no headers are sent, MS returns the minimum status code listed in the spec (200 if no codes were defined).

The idea of MS2-Status-Code is not overriding the default but just filter the listed codes

Reply:
Fixed with the table, I hope.